localstack/services/sns/sns_listener.py

```diff
@@ -524,7 +524,7 @@ def create_sqs_message_attributes(subscriber, attributes):
             'DataType': value['Type']
         }
         if value['Type'] == 'Binary':
-            attribute['BinaryValue'] = value['Value']
+            attribute['BinaryValue'] = bytearray(value['Value'])
```

Comment:
Thanks for this fix @gruff4l0 ! To be on the safe side - in case `value['Value']` happens to be string instead of binary type, can we import `to_bytes` from `utils/common.py` at the top of the file, and then add this conversion here:
```
attribute['BinaryValue'] = bytearray(to_bytes(value['Value']))
```

Reply:
hey @whummer , thanks for the suggestion. 

I have tested it on my use case with java app, and, unfortunately, this exact change breaks expected functionality. 
If there is case when we can play safe, I am more than happy to give it a try, but as for now, it looks like safe casting is not the right one. 

My understanding is so far, that sqs client expects raw bytes, and we feeding it with already decoded bytes. It tastefully takes it all, but the digest result is not as expected.

Reply:
Thanks for the update @gruff4l0 . You mentioned that this change breaks expected functionality - can you please share the mismatch or error message you're seeing? Thanks

My comment above was addressing the following issue: 
```
$ python -c 'print(bytearray(b"bytes"))'
bytearray(b'bytes')
$ python -c 'print(bytearray("string"))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: string argument without an encoding
$ python -c 'print(bytearray("string".encode("UTF-8")))'
bytearray(b'string')
```

> Also, I think value['Value'] always is a string

If that's the case (i.e., if the value is a `str` and not `bytes`), then I think the `to_bytes(..)` conversion above should be required. Or am I missing something?

Reply:
> can you please share the mismatch or error message you're seeing?

I described it details in the corresponding ticket, in the https://github.com/localstack/localstack/issues/2432#issuecomment-640158726

As for `str` and `bytes`, let me clarify that argument is always a string:
The argument is always _encoded_ string (in your example, the third case `"string".encode("UTF-8")`). Apologies, I misunderstood your initial note. 
The fact that wrongly delivered binary payload - is the correct payload, that got encoded the second time shows the direction of the fix. 

In my understanding the flow is the following: 
1) The client sends the request to SNS with aws library that makes encoding of binary message attribute;
2) LocalStack gets the request and converts messages from SNS to SQS with boto library by copying message attributes; But at this point, the binary attribute is already base64 encoded string.
3) LocalStack sends SQS requests with boto, and the boto library makes the encoding of binary attributes again.

To avoid second encoding, binary attributes should be set as bytearray during SNS message to SQS message conversion.

Reply:
Ok, thanks again for the update and for describing the detailed flow.

> The argument is always encoded string (in your example, the third case "string".encode("UTF-8")).

If that's the case, then I still don't fully understand why the change I suggested breaks the expected functionality, as you mentioned:
```
attribute['BinaryValue'] = bytearray(to_bytes(value['Value']))
```

The change above would fix case (2) AND cover both cases (1) and (3) above:
* Case (1) - if `value['Value']` is already an encoded string, then `to_bytes` simply returns the unchanged value, and it works
* Case (2) - if `value['Value']` is an unencoded string, then `to_bytes` would convert it to an encoded string, and it works
* Case (3) - if `value['Value']` is already an encoded string, then `to_bytes` simply returns the unchanged value, and it works (this case is actually identical to (1))

My concern is still that the current solution does not cover case (2) from above, if potentially `value['Value']` is an (unencoded) string. But I don't want to get into too much nitpicking here :) , so I think we can simply merge the PR as is. We can open follow-up issues and/or PRs as needed. Thanks again